code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -259,31 +272,46 @@
 		T = check
 	return (get_line(src, T) - get_turf(src))
 
+/**
+ * Returns a list of open turfs that allow gas to pass in a line.
+ * Arguments:
+ * * atom/target - The target for the line.
+ * * range - The maximum number of turfs it can return.
+*/
+/mob/living/simple_animal/hostile/space_dragon/proc/get_stream_turfs(atom/target, range)
+	var/list/turfs = list()
+	for(var/turf/turf in line_target(0, range, target))
+		if(isclosedturf(turf))
+			return turfs
+		if(isspaceturf(turf))
+			for(var/atom/atom in turf)
+				if(!atom.can_atmos_pass(turf))
+					return turfs
+		else if(!TURF_SHARES(turf))
+			return turfs
+		turfs += turf
+	return turfs
```

Comment:
this will fail if you breath into a 1x1 room

Reply:
Is there a cheap way of handling that?

Reply:
rather then checking TURF_SHARES
store the PREVIOUS turf we walked over, and use CAN_SHARE or whatever it is between those two
If there is no previous turf, ignore the check